src/authorization-response/AuthorizationResponse.ts

```diff
@@ -115,14 +115,16 @@ export class AuthorizationResponse {
 
     if (hasVpToken) {
       const wrappedPresentations = await extractPresentationsFromAuthorizationResponse(response, { hasher: verifyOpts.hasher });
+      const presentationSubmission =
+        responseOpts.presentationExchange?.presentationSubmission ?? authorizationResponsePayload.presentation_submission;
 
       await assertValidVerifiablePresentations({
         presentationDefinitions,
         presentations: wrappedPresentations,
         verificationCallback: verifyOpts.verification.presentationVerificationCallback,
         opts: {
           ...responseOpts.presentationExchange,
-          presentationSubmission: responseOpts.presentationExchange?.presentationSubmission ?? authorizationResponsePayload.presentation_submission,
+          ...(presentationSubmission ? { presentationSubmission: presentationSubmission } : {}),
```

Comment:
This is wrong! If the above hasVpToken is true, there needs to be a submission. So either the logic to determine the hasVpToken is wrong, or something else is happening. This PR is wrong for sure

Reply:
I found the submission in authorizationResponsePayload.presentation_submission not in presentationExchange. So this needs to be fixed elsewhere then.
So that that makes this PR is not to merge, just for diagnostics. I will revert the change in my branch as soon as presentationSubmission appears in presentationExchange

Reply:
No you are likely simply misunderstanding how it works. Presentation Exchange is a service that provides the submission,  but only when a vp_token is needed

Reply:
I have found the problem, in the SSI SDK version I am using with RSA support, presentationSubmission in OPSession is not loaded yet.  But this line still has to be in opts
```
          ...(presentationSubmission ? { presentationSubmission: presentationSubmission } : {}),
```
It's asserted in assertValidVerifiablePresentations
```
  } else if (args.presentationDefinitions && !args.opts.presentationSubmission) {
    throw new Error(`No presentation submission present. Please use presentationSubmission opt argument!`);
  } else if (args.presentationDefinitions && presentationsWithFormat) {

```

Reply:
No that line should not be in opts

Reply:
Ah I see now it has the ... before responseOpts.presentationExchange so it should be in opts already.  So the problem was my OPSession which was from before October 13th and I patched before it up by getting it from elsewhere.
I reverted that code